### DISCUSSION TITLE
Cohabitation between chakra v2 and v3 ?

### DISCUSSION THREAD

**Discussion by vicvytruve** [chakra-ui/chakra-ui#9158] [Q&A] [answered]
Hello,

We are using chakra v2 in our project and planned to switch to v3. We quickly discovered that migrating the whole app in one go will be very long and tedious four our team. Lots of Chakra components have been changed in there implementation, and we don't have the taskforce to to everything at once.

Is there a possibility of having both versions of chakra UI at the same time on our React app ? So we can use V3 for new developments and take the time to switch legacy code ?

Can both Providers work at the same time ?

Thank you

**Comment by Daidalos117** [comment]
I don't think it's possible to use both. But I agree that there is so many breaking changes. Shame they didn't create codemod migration as Next.js is using. I know it would be more work for the authors, but it would save thousands of hours for others. I even thought about creating it by myself, but I am mostly done with manual migration now, so there is not much reason for me now.

**Comment by segunadebayo** [comment, marked as answer]
We do not support running v2 and v3 in the same app. Migrate fully to v3.
